### ISSUE TITLE
dynamic import catch handler not firing on failure

### ISSUE THREAD

**Issue by dnalborczyk** [rollup/rollup#2957]
- Rollup Version: v1.15.6, v1.16.1
- Operating System (or Browser): n/a
- Node Version: v12.4.0

### How Do We Reproduce?

```js
import('./foo.js').catch((e) => {
  // catch error here
});
```

when bundled to commonjs:
```js
// commonjs
Promise.resolve(require('./foo.js')).catch((e) => {
  // catch error here
  // will never execute if ./foo.js doesn't exist
})
```

`Promise.resolve` takes only a value, not an expression. I think `rollup` should either use:
```js
Promise.resolve().then(() => require('./foo.js'))
```
or the Promise constructor:
```js
new Promise((res) => res(require('./foo.js')))
```

**Comment by lukastaegert** [comment]
Good point. Yes, an import error would not be caught be Promise.resolve.

**Comment by lukastaegert** [comment]
Fix at #2958

**Comment by dnalborczyk** [comment]
@lukastaegert  awesome, thanks!!
